src/rgw/rgw_file.h

```diff
@@ -802,7 +802,10 @@ namespace rgw {
 			     const uint32_t flags = RGWFileHandle::FLAG_NONE) {
       using std::get;
 
-      LookupFHResult fhr { nullptr, RGWFileHandle::FLAG_NONE };
+      // This seems like a Clang compiler error.
+      // It result in irgw::RGWFileHandle::FLAG_NONE being undefined during linking.
+      // LookupFHResult fhr { nullptr, RGWFileHandle::FLAG_NONE };
+      LookupFHResult fhr { nullptr, 0 };
```

Comment:
> But what I do not understand is that there are more references to RGWFileHandle::FLAG_NONE ( for instance a few lines earlier ),

yeah because `constexp` *and* passing by rvalue.

> And per your link-text, that static declaration would require an explicit definition.
> But that sort of surpasses the constexpr declaration.

please note the difference of ODR-use and non ODR-use.

> Interested in what your suggestion for a fix is.

```diff
-      LookupFHResult fhr { nullptr, RGWFileHandle::FLAG_NONE };
+      LookupFHResult fhr { nullptr, uint32_t(RGWFileHandle::FLAG_NONE) };
```